test/src/test/java/hudson/model/HudsonTest.java

```diff
@@ -79,6 +82,64 @@ public void globalConfigRoundtrip() throws Exception {
         assertEquals(8, j.jenkins.getNumExecutors());
     }
 
+    @Test
+    public void globalConfigAllowedWithConfigurePermission() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+               .grant(Jenkins.CONFIGURE_JENKINS, Jenkins.READ).everywhere().toEveryone());
+
+        HtmlForm form = j.createWebClient().goTo("configure").getFormByName("config");
+        HtmlPage updated = j.submit(form);
+        assertEquals("User with CONFIGURE_JENKINS permission should be able to update global configuration",
+                     200, updated.getWebResponse().getStatusCode());
+    }
+
+    @Test
+    public void someGlobalConfigurationIsNotDisplayedWithConfigurePermission() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                                                   .grant(Jenkins.CONFIGURE_JENKINS, Jenkins.READ).everywhere().toEveryone());
+
+        HtmlForm form = j.createWebClient().goTo("configure").getFormByName("config");
+        String formText = form.asText();
+        assertThat("Shouldn't be able to configure # of executors", formText, not(containsString("executors")));
+        assertThat("Shouldn't be able to configure Global properties", formText,
+                   not(containsString("Global properties")));
+        assertThat("Shouldn't be able to configure Administrative monitors", formText, not(containsString(
+                "Administrative "
+                + "monitors")));
+        assertThat("Shouldn't be able to configure Shell", formText, not(containsString("Shell")));
+    }
+
+    @Test
+    public void someGlobalConfigCanNotBeModifiedWithConfigurePermission() throws Exception {
+        int currentNumberExecutors = j.getInstance().getNumExecutors();
+        View primary = j.getInstance().getPrimaryView();
+        String shell = getShell();
+
+        j.getInstance().addView(new ListView("otherView"));
+
+        //Load the form without authz restrictions
+        HtmlForm form = j.createWebClient().goTo("configure").getFormByName("config");
+        form.getInputByName("_.numExecutors").setValueAttribute(""+(currentNumberExecutors+1));
+        form.getSelectByName("primaryView").setSelectedAttribute("otherView", true);
+        form.getInputByName("_.shell").setValueAttribute("/fakeShell");
+
+        //User has CONFIGURE_JENKINS permission only, so should't be allowed to post a change on number of executors
```

Comment:
Maybe this should move to the top to match the given-when-then

Reply:
What I'm trying to test is that the POST request is not allowed.
I can't move it to the top, because I need to render the form before I'm restricting the permissions.
If I restrict the permissions on the top, I wouldn't be able to get that form, and I need it to recreate the POST request. 

(I'm not sure if I'm explaining correctly :smile: )

Reply:
I'm trying to explain it better in the comments